pandas/core/arrays/numeric.py

```diff
@@ -199,3 +201,48 @@ def reconstruct(x):
             return tuple(reconstruct(x) for x in result)
         else:
             return reconstruct(result)
+
+    def _apply(self, func: Callable, **kwargs) -> NumericArray:
+        """
+        Applying a given function on the array
+
+        Parameters
+        ----------
+        func : callable function to apply
+        **kwargs
+            additional arguments for the function if necessary
+
+        Returns
+        -------
+        NumericArray
+            The image of the NumericArray
+        """
+        values = func(self._data, **kwargs)
+        return type(self)(values, self._mask.copy())
```

Comment:
if func returns a view or is a no-op, would we need to the the same for the mask?

Reply:
It was related to a comment of @jorisvandenbossche in the original PR proposal #38866

> The mask needs to be copied I think? (result should not share a mask with the original array, because otherwise editing one can modify the other. We should probably also test this)

Here, I think this method should only be used with functions which return a copy (which is the case of most numpy functions I know --- without the `out` argument). It may need to be précised in the docstring for future developpers.

Reply:
Agreed, see comment above as well. Let's not generalize this prematurely, just knowing (and documenting) the `func` returns a new object is fine for now IMO.